src/main/java/com/cloudbees/groovy/cps/tool/Translator.java

```diff
@@ -138,8 +137,9 @@ public Translator(CompilationTask task) throws IOException {
     /**
      * Transforms a single class.
      */
-    public void translate(String fqcn, String outfqcn, Predicate<ExecutableElement> methodSelector) throws JClassAlreadyExistsException {
+    public void translate(String fqcn, String outfqcn, Predicate<ExecutableElement> methodSelector, String sourceJarName) throws JClassAlreadyExistsException {
         final JDefinedClass $output = codeModel._class(outfqcn);
+        $output.annotate(Generated.class).param("value", Translator.class.getName()).param("date", new Date().toString()).param("comments", "based on " + sourceJarName);
```

Comment:
Helpful hint to anyone looking at the generated source file and wondering who on earth wrote all that!